### DISCUSSION TITLE
customizing controls

### DISCUSSION THREAD

**Discussion by zprofreelancer** [serenity-is/Serenity#5383] [Help] [answered]
hello,
i need to edit the country and city and representatives controls of the customer page. i need to change the text of the controls......how to do that please. also how to change the representative input text to dropdown control.

Thanks.

**Comment by erehtolleh** [comment, marked as answer]
I don’t know Serenity well but a lot of your questions can be found in the manual.  

You can change most fields through attributes on xyzrow.cs or xyzforms.cs. Row applies everywhere and form is just in the form. Any other changes could be done through typescript or write your own Editors. Like the sample Phone Editor. Look in the Wiki, has lots of helpful use cases. 

Here is a link to what you are looking for. 
[Changing Attributes](https://tinyurl.com/y2xfkjby)

**Reply by zprofreelancer** [reply]
Thanks alot... i also found this useful link https://serenity.is/docs/entities/mapping_attributes

**Comment by zprofreelancer** [comment]
sorry to ask....but how to choose only country and city and representatives to be displayed on the customer page?...also the click action on "new customer" button that opens the customer form page where is that done?

**Reply by JohnRanger** [reply]
Hi @zprofreelancer ,
I think that you really have to first work yourself through the excellent Serenity Guide: https://volkanceylan.gitbooks.io/serenity-guide/content/ - Especially do the Movie Database tutorial - This will give you a very good understanding of the basics (which you are asking here) :-)

After that, I suggest that you look through the also excellent Wiki: https://github.com/serenity-is/Serenity/wiki. There are over 140 answers to more complex challenges.

And if - after doing all your above homework - you still have unanswered questions, please feel free to ask here :-)

With kind regards,

John
